Fix issue 87

l0n3star opened the pull request:
Issue 87:  Fix practicalplants name properties to start with upper case letter

petteripitkanen commented on db/practicalplants.js:
```diff
@@ -105,6 +105,19 @@ function readCrops() {
 
+    for (const property of NAME_PROPERTIES) {
+      if (object[property]) {
+        if (/[a-z]/.test(object[property][0])) {
```
It is probably safer to use `letter === letter.toLowerCase()` for possible non-ASCII letters.

petteripitkanen replied:
Even better `character !== character.toUpperCase()` for possible other characters.

petteripitkanen replied:
Never mind, it is okay to limit the replace to only `/[a-z]/`. In the test it was already checked that these properties always start with `/[a-zA-Z]/`.

l0n3star commented:
Sorry about modifying package-lock.json.  I have reverted it.

I updated code to log erroroneous objects to stderr.  Only 1 object found:

{"binomialName":"Phaseolus vulgaris","commonName":"common bean, string bean, field bean, flageolet bean, French bean, garden bean, haricot bean, pop bean, snap bean","poorNutrition":"false","hardinessZone":10,"soilTexture":["sandy","loamy","clay"],"soilPh":["neutral","alkaline"],"soilWaterRetention":["well drained"],"shade":"no shade","sun":"full sun","water":"moderate","drought":"intolerant","ecosystemNiche":[],"lifeCycle":["annual"],"herbaceousOrWoody":null,"deciduousOrEvergreen":null,"growthRate":null,"matureMeasurementUnit":null,"matureHeight":2,"matureWidth":null,"flowerType":"hermaphrodite","pollinators":[],"wind":"false","maritime":"false","pollution":"false","functions":["nitrogen fixer"],"growFrom":[],"cuttingType":[],"fertility":["self fertile"],"rootZone":null,"family":"Leguminosae","genus":"Phaseolus","salinity":null}

l0n3star commented:
Thanks for feedback.  Would it be better to log to a file?  Open to
suggestions.

On Sun, Aug 25, 2019 at 11:45 AM petteripitkanen <notifications@github.com>
wrote:

> *@petteripitkanen* approved this pull request.
>
> Thanks, this fixes #87
> <https://github.com/Ecohackerfarm/powerplant/issues/87>, and is related
> to #31 <https://github.com/Ecohackerfarm/powerplant/issues/31>.
>
> The logging in readCrops is okay for now as this function is used only
> for the migration process and tests (for tests it can be disabled with npm
> run test 2>/dev/null), but will probably be modified during #89
> <https://github.com/Ecohackerfarm/powerplant/issues/89> when more
> abstract modules are created for managing the Crop and PracticalplantsCrop
> types.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ecohackerfarm/powerplant/pull/88?email_source=notifications&email_token=ALAQO35UVVBORT43KX4OYILQGLHK3A5CNFSM4IPEWWXKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCTIJGY#pullrequestreview-279348379>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALAQO35A77Z2SDLMLA7NIYTQGLHK3ANCNFSM4IPEWWXA>
> .
>

petteripitkanen commented:
It could be clearer to log in the tests, but only if `ENABLE_LOGGING` is `true`. Then this constant would be committed as `false` and used only locally when needed.

l0n3star commented:
Ah I see.  Good idea.

On Sun, Aug 25, 2019 at 12:55 PM petteripitkanen <notifications@github.com>
wrote:

> It could be clearer to log in the tests, but only if ENABLE_LOGGING is
> true. Then this constant would be committed as false and used only
> locally when needed.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ecohackerfarm/powerplant/pull/88?email_source=notifications&email_token=ALAQO37MC6FGWWQYCRNHNBTQGLPUVA5CNFSM4IPEWWXKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5C2RLY#issuecomment-524658863>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALAQO36MCN4EUHYYROPR62LQGLPUVANCNFSM4IPEWWXA>
> .
>
